servers/rendering/renderer_rd/environment/gi.h

```diff
@@ -410,7 +410,7 @@ class GI : public RendererGI {
 			int32_t image_size[2];
 
 			int32_t world_offset[3];
-			uint32_t sky_mode;
+			int32_t sky_mode;
```

Comment:
I would rename this to `sky_flags` then pack the sky mode in the first two bits and the w channel of the quaternion in the third bit. 

Your packing is more clever and requires fewer instructions to unpack. But I think it will be more confusing for future readers of the code and using bit packing makes everything more explicit.